### PR TITLE
HTTP/2: add exception error messages and log them

### PR DESCRIPTION
**Pull request by cesarbs**
TODO:

- [x] Actually log exceptions
- [x] Verify exceptions in tests

### CODE REVIEW

**Comment by cesarbs** [comment]
This is ready for review.

File: src/Kestrel.Core/Internal/Http2/Http2Connection.cs [hunk 1]
```diff
@@ -589,6 +604,7 @@ private Task DecodeHeadersAsync(bool endHeaders, Span<byte> payload)
             }
             catch (Http2StreamErrorException ex)
             {
+                // TODO: log
```
Comment (halter73) [inline src/Kestrel.Core/Internal/Http2/Http2Connection.cs]: Why not do this a part of this PR?

Comment (cesarbs) [reply src/Kestrel.Core/Internal/Http2/Http2Connection.cs]: Because I totally forgot about it 😆

**Comment by cesarbs** [comment]
Updated w/ logs for `Http2StreamErrorException`.

I've also done one minor change, which is to always close the connection when receiving a WINDOW_UPDATE frame. You can see the change here, as well as the explanation for it in the comments: https://github.com/aspnet/KestrelHttpServer/pull/2108/commits/1f2b55587504bfc0a9d384d66a263434a752d61a#diff-4ab77515238e5e964ae9a3a4615205b1R553

**Pull request merged**